### PR TITLE
feat(generator): app starter

### PR DESCRIPTION
**Pull request by jsomsanith-tlnd**
**What is the problem this PR is trying to solve?**
`talend:app` generator doesn't generate anything except an index.

**What is the chosen solution to this problem?**
Generate an app based on parameters

Base :
- folder structure
- chore using `@talend/scripts`
- bootstrap via `cmf.bootstrap`
- `focus-visible` polyfill
- `service` example
- `component` (using the service) example

Option i18n 
- i18n via `i18next`
- extraction via `i18next-scanner`
- i18n process via `talend-i18n.json` config

Option cmf
- settings injection title and mount action
- register action creators and settings fetch via `cmf.bootstrap`
- `cmfConnect` with action creation dispatch

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/generator/generators/app/templates/src/app/services/dataset/dataset.service.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+export const reducer = {
+	[DATASET_REDUCER_KEY]: datasetReducer,
+};
+
+export function fetchDatasets() {
+	return dispatch => {
+		dispatch({ type: DATASET_SET_FETCHING, isFetching: true });
+		setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/generator/generators/app/templates/src/app/services/dataset/dataset.service.js]: To simulate a fetch delay

File: packages/generator/generators/app/templates/src/app/index.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+<html>
+    <head>
+        <meta charset="utf-8">
+        <meta http-equiv="x-ua-compatible" content="ie=edge">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <link rel="icon" href="/favicon.png" />
+    </head>
+    <body>
+        <!--[if lt IE 8]>
+            <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</p>
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/generator/generators/app/templates/src/app/index.html]: talend/scripts doesn't provide an html template. The app provides it

File: packages/generator/generators/app/templates/src/app/services/home/home.service.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	[HOME_REDUCER_KEY]: homeReducer,
+};
+
+export function toggleSidePanel() {
+	return { type: HOME_SIDEPANEL_TOGGLE };
+}
+
+export function isSidePanelDocked(state) {
+	return state[HOME_REDUCER_KEY].sidePanelDocked;
+}
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/generator/generators/app/templates/src/app/services/home/home.service.js]: It's an example, more elements can be managed by Home service : another drawer, an active tab, etc. The idea is not to do a service per component.

**Comment by jsomsanith** [comment]
The idea is not to generate a settings centric app. The other generator (`talend:react-cmf`) is settings centric.
And as it should be the starting point of new projects, I thought that it would be good to add common things like i18n

File: packages/generator/generators/app/templates/src/app/services/dataset/dataset.service.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+			fetch('/datasets.json')
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/generator/generators/app/templates/src/app/services/dataset/dataset.service.js]: You're right 👍

**Comment by frassinier** [comment]
@jsomsanith and @jmfrancois Do we still need this?

**Comment by jsomsanith-tlnd** [comment]
Don't know, and I guess it's outdated now.
Let's not put more energy maintaining a starter that will be deprecated soon anymore.

**Pull request closed without merging**